[CMS-477] Investigate fixed image dimensions for tab group, update Image component

levadadenys opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->
### [CMS-477] Investigate fixed image dimensions for tab group, update Image component

* [feat(cms/image): update image to use figure element as a container, i…](https://github.com/code-dot-org/code-dot-org/pull/64851/commits/4d88a741da9cfb0002caddcfacd4a3551288a642)
* [feat(image): make cms/image component a general component-library com…](https://github.com/code-dot-org/code-dot-org/pull/64851/commits/45ed52acedbb403d0751ca2eed002a0820512b70)
* [feat(cms/tabGroup): update image styles](https://github.com/code-dot-org/code-dot-org/pull/64851/commits/54c7d296de49826c4f181f404a02099a102582f2)
* [chore(image): minor stories update](https://github.com/code-dot-org/code-dot-org/pull/64851/commits/de8493f6fdbaa1253938911115b1f4a5017eca57)

https://github.com/user-attachments/assets/2046230f-1870-41af-a3cf-1108f43ea216


## Links
- jira ticket: [CMS-477](https://codedotorg.atlassian.net/browse/CMS-477)

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

<!--
- spec: []()
-->

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

kelbyhawn commented on frontend/packages/component-library/src/cms/tabGroup/tabGroup.module.scss:
```diff
@@ -45,10 +45,10 @@ $mobileBreakpoint: 1024px;
   width: 28.6699rem;
-
-  border-radius: variables.$regular-border-radius;
+  height: 18.7234rem;
```
Is this a magic number or what we always want this to be? (Same question for line 92)

stephenliang replied:
+1 could you explain the value?

levadadenys replied:
These are not magic numbers. Unless any custom styles are added, we always want the images in TabGroup to have these values. These values are just from Figma. The image will always have a fixed width and height based on Figma.

I have to admit I was a little bit confused by decimal values, but that's rems, and that's ok and approved by @moshebaricdo

kelbyhawn replied:
Ok cool, that makes sense! I assume we'll do the same for ActionBlock too?

stephenliang replied:
Could you add a comment explaining this and where to get the numbers from should they update?

levadadenys replied:
> Could you add a comment explaining this and where to get the numbers from should they update?

I think that's redundant, as it is practically the same as if we would write a comment for any other line of styles as it's the same thing

levadadenys replied:
> Ok cool, that makes sense! I assume we'll do the same for ActionBlock too?

Absolutely, if image has fixed dimensions in figma, we should set them to the component as well

kelbyhawn replied:
If there's not much visual difference could we trim the decimals and just round these to like `18.7` instead? (To address the need for a comment and confusion about why there are so many decimals 😅)

levadadenys replied:
We have a linter rule for that and this value is already shortened to fit it. Trim more decimals may give unexpected regressions, so I'll just add a comment about where these values are taken from and why as you suggested above.

But in a follow-up PR. :) I will merge this PR now as is not to block anyone

kelbyhawn replied:
Sounds good, ty! I made a [ticket](https://codedotorg.atlassian.net/browse/CMS-491) to update this for the ActionBlocks too I can do this sprint 🙌

kelbyhawn commented on frontend/packages/component-library/src/image/Image.tsx:
```diff
@@ -0,0 +1,93 @@
+  /** Optional inline styles. You can either pass it as a prop, or use custom className to set the needed styles values.
+   Using custom className is recommended, but we also support setting it via prop for the cases when it might be
+   impossible to use custom className. */
+  style?: CSSProperties;
```
Cool! ✨

levadadenys replied:
Oh, thanks for noticing this! I actually plan to remove this `style` and `height` and `width` from prop types :)  as I don't think we need this. However, inline styles will still be supported alongside with other native htmlelement props, meaning you won't see it in type definition and storybook, but there still will be an option to pass inline styles object prop if it's really required

kelbyhawn reviewed (approved):
Looks incredible!! Thanks for making these updates 🎉

levadadenys commented:
Updated pr, removed `width` and `height` support from the component as it won't work properly and might be confusing for the current implementation.
inline styles `style` prop is still supported and these inline styles are being applied to `figure` element. I just removed this prop from proptypes directly to avoid encouraging people to use inline styles and make it more consistent with all the other component we have. (most of them do support inline styles, they just don't announce it though storybook, but you can still see them through typescript autocomplete if you try)
